### PR TITLE
Fix blocking commands moved error

### PR DESCRIPTION
**Pull request by BorysTheDev**
We had incorrect results for blocking commands for KEY_MOVED error.

### CODE REVIEW

File: src/server/list_family.cc [hunk 1]
```diff
@@ -1226,8 +1226,8 @@ void ListFamily::BPopGeneric(ListDir dir, CmdArgList args, ConnectionContext* cn
     case OpStatus::TIMED_OUT:
       return rb->SendNullArray();
     case OpStatus::KEY_MOVED:
-      // TODO: proper error for moved
-      return cntx->SendError("-MOVED");
+      // if key is moved the cluster is definitely exist and we must get an error
+      return cntx->SendError(*cluster::SlotOwnershipErrorStr(*transaction->GetUniqueSlotId()));
```
Comment (chakaz) [inline src/server/list_family.cc:1230]:
```suggestion
      auto error = cluster::SlotOwnershipErrorStr(*transaction->GetUniqueSlotId());
      CHECK(error.has_value());
      return cntx->SendError(*error);
```

File: src/server/zset_family.cc [hunk 2]
```diff
@@ -1364,6 +1364,9 @@ void BZPopMinMax(CmdArgList args, ConnectionContext* cntx, bool is_max) {
     case OpStatus::CANCELLED:
     case OpStatus::TIMED_OUT:
       return rb->SendNullArray();
+    case OpStatus::KEY_MOVED:
+      // if key is moved the cluster is definitely exist and we must get an error
+      return cntx->SendError(*cluster::SlotOwnershipErrorStr(*transaction->GetUniqueSlotId()));
```
Comment (chakaz) [inline src/server/zset_family.cc:1369]: ditto

File: tests/dragonfly/cluster_test.py [hunk 3]
```diff
@@ -787,6 +787,47 @@ async def test_cluster_blocking_command(df_server):
     await close_clients(c_master, c_master_admin)
 
 
+@dfly_args({"proactor_threads": 4, "cluster_mode": "yes"})
+async def test_blocking_comands_cancel(df_factory, df_seeder_factory):
```
Comment (chakaz) [inline tests/dragonfly/cluster_test.py:791]:
```suggestion
async def test_blocking_commands_cancel(df_factory, df_seeder_factory):
```

**Review by chakaz — COMMENTED** [review]
Small nits, overall looks good

File: src/server/list_family.cc (hunk 1 above)
Comment (BorysTheDev) [reply src/server/list_family.cc:1230]: We already know that we have an error. I don't think that additional check is needed

Comment (chakaz) [reply src/server/list_family.cc:1230]: A `CHECK` _is_ a way to ensure that your assumptions are correct.
It behaves better in that you get better stack frames and no memory corruption and random side effects if your assumptions are incorrect.
It's also a way to "document" your assumptions in a better and more concise way than a comment

File: src/server/main_service.cc [hunk 4]
```diff
@@ -958,16 +958,10 @@ optional<ErrorReply> Service::CheckKeysOwnership(const CommandId* cid, CmdArgLis
...
-  }
-
-  if (keys_slot.has_value() && !cluster_config->IsMySlot(*keys_slot)) {
-    // See more details here: https://redis.io/docs/reference/cluster-spec/#moved-redirection
-    cluster::ClusterNodeInfo master = cluster_config->GetMasterNodeForSlot(*keys_slot);
-    return ErrorReply{absl::StrCat("-MOVED ", *keys_slot, " ", master.ip, ":", master.port)};
+  if (keys_slot.has_value()) {
+    if (auto error_str = cluster::SlotOwnershipErrorStr(*keys_slot); error_str) {
+      return ErrorReply{std::move(*error_str)};
+    }
```
Comment (dranikpg) [inline src/server/main_service.cc:964]: you can just return an error reply directly, will simplify the code around 🙂

Comment (BorysTheDev) [reply src/server/main_service.cc:964]: we don't need it in other places

**Review by chakaz — APPROVED** [review]

**Pull request merged**